xapian-core/bin/xapian-progsrv.cc

```diff
@@ -71,7 +71,13 @@ int main(int argc, char **argv)
 		cout << PROG_NAME " - " PACKAGE_STRING << endl;
 		exit(0);
 	    case 't':
-		timeout = atoi(optarg) * 1e-3;
+		unsigned int temp;
+		if (!parse_unsigned(optarg, temp)) {
+		    cout << "timeout must be a positive integer" << endl;
+		    show_usage();
+		    exit(0);
```

Comment:
We should `exit(1)` here (the `--help` handling above does `show_usage()` then `exit(0)` because that's not an error case).

Reply:
fixed in a fixup commit.